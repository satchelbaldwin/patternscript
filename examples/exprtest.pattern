pattern x = {
    iteration_type = time;
    length = 1.0 seconds;
    actions = {
        origin = (1, 2) + 3;
    }
}