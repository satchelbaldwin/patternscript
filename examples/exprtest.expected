Head(
    HeadData { 
        definitions: {
            "x": Pattern(
                PatternData { 
                    block: Block { 
                        definitions: {
                            "actions": Block(
                                Block { 
                                    definitions: {
                                        "origin": Expr(Binary(Add, Int(2), Int(1)))}, 
                                        statements: [] 
                                    }
                                ), 
                            "iteration_type": Variable("time"), 
                            "length": Time(Time(Float(1.0)))
                        }, 
                        statements: [] 
                    } 
                }
            )
        } 
    }
)
